syntax="proto3";

package client;

message init_round {
    int32 round_number = 1;
}

message metrics_results {
    float accuracy = 1;
    float precision = 2;
    float recall = 3;
}

message models_weights {
    string model_name = 1;
    bytes weights = 2;
}

message void {
}

service api {
    rpc train_model(void) returns (void);
    rpc connect(void) returns (void);
}

